SharpMap/Rendering/Decoration/ScaleBar/ScaleBarAltUnit.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+
+namespace SharpMap.Rendering.Decoration.ScaleBar
+{
+    /// <summary>
+    /// Scale Bar map decoration with switching between units for large and small scales
+    /// </summary>
+    [Serializable]
+    public class ScaleBarAltUnit : ScaleBar
```

Comment:
Why do you derive this class and don't extend `ScaleBar`? If `BarUnitAlt1` and `BarUnitAlt2` are of same value, this class behaves just like old `ScaleBar`, doesn't it?

Reply:
I started modifying existing class, but then thought an inherited class was a cleaner approach. I can now see there is a bit of an issue with existing projects having to change `ScaleBar` to `ScaleBarAltUnit` rather than just accessing new props, so happy to re-visit this if you would prefer existing class to be modified.

Yes, it behaves just like old `ScaleBar` when `BarUnitAlt1` = `BarUnitAlt2`. 

I should probably set default value for `BarUnitAlt2` = **km** so it is more obvious and "works" immediately when added to map.

Reply:
Have posted update setting defaults for BarUnitAlt1 = m and BarUnitAlt2 = km, and modified comment on BarUnit.

Reply:
Please don't override `ScaleBar`, extend it